resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/session/TestBug297635.java

```diff
@@ -45,26 +53,12 @@ public BundleContext getContext() {
 		return Platform.getBundle(PI_RESOURCES_TESTS).getBundleContext();
 	}
 
-	public void test1() {
+	public void testBug() throws Exception {
```

Comment:
The method becomes much easier to read and comprehend due to this separation of functionality into dedicated methods.
Maybe you could even remove the comments and, if necessary, improve the method names. That could further improve comprehensibility, also w.r.t. the method actually being composed of two chunks that made up the original test methods (install / reinstall).

Reply:
You mean:
1. Rename `testBug` to `testSavedStatesAreDiscardedAfterSnapshot` (or do you have another name in mind?)
2. Remove comment (`/* ... */`) in `reinstallBundle`

If not, could you please give me some example(s)?

Reply:
Sorry for being imprecise: I just meant to remove the comments within the `testBug` method, i.e., the comments just repeating what the method called in the line after the comment is doing, like
```java
// install a bundle
installBundle();
```
The method of the test method can remain as it is.

Reply:
I get it now, thanks.

I created/renamed some private methods and removed unnecessary comments in `testBug` in 0347088028c67a0517c11c530f796d077076afbe